index.html

```diff
@@ -101,212 +220,1289 @@ <h3>
+
+        <ol>
+          <li>
+            Let |stream:ReadableStream| be a [=new=] {{ReadableStream}}.
+          <li>
+            Let |pullAlgorithm| be the following steps:
+            <ol>
+              <li>
+                Let |desiredSize| be the
+                <a href="https://streams.spec.whatwg.org/#desired-size-to-fill-a-streams-internal-queue">desired size</a>
+                of [=this=].{{TCPSocket/[[readable]]}}'s
+                <a href="https://streams.spec.whatwg.org/#internal-queues">internal queue</a>.
+              <li>
+                Run the following steps in parallel:
+              <ol>
+                <li>
+                  Invoke the operating system to read up to |desiredSize| bytes from the socket, placing the result in the [=byte sequence=] |bytes|.
+                <li>
+                  [=Queue a global task=] on the [=relevant global object=] of
+                  [=this=] using the [=TCP socket task source=] to run the
+                  following steps:
+                  <ol>
+                    <li>
+                      If the connection was closed gracefully, run the following steps:
+                      <ol>
+                        <li>
+                          Invoke [=ReadableStream/close=] on [=this=].{{TCPSocket/[[readable]]}}.
+                        <li>
+                          Invoke the steps to [=handle closing the TCP readable stream=].
+                      </ol>
+                      <div class="note">
+                        This is triggered by the peer sending a packet with the FIN flag set
+                        and is typically indicated by the operating system returning 0 bytes
+                        when asked for more data from the socket.
+                      </div>
+                    <li>
+                      If no errors were encountered run the following steps:
+                      <ol>
+                        <li>
+                          Let |buffer| be a [=new=] {{ArrayBuffer}} created from |bytes|.
+                        <li>
+                          Let |chunk| be a [=new=] {{Uint8Array}} view over |buffer|,
+                          who's length is the length of |bytes|.
+                        <li>
+                          Invoke [=ReadableStream/enqueue=] on
+                          [=this=].{{TCPSocket/[[readable]]}} with |chunk|.
+                      </ol>
+                    <li>
+                      If a network or operating system error was encountered, invoke
+                      [=ReadableStream/error=] on [=this=].{{TCPSocket/[[readable]]}} with
+                      a "{{NetworkError}}" {{DOMException}} and invoke the steps
+                      to [=handle closing the TCP readable stream=].
+                </ol>
+              <li>
+                Return [=a promise resolved with=] `undefined`.
```

Comment:
This needs to be unindented one more level (outside the "in parallel" steps).

Reply:
Fixed.